services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontSignedCookie.java

```diff
@@ -0,0 +1,251 @@
+     */
+    public String signatureKey() {
+        return SIGNATURE_KEY;
+    }
+
+    /**
+     * Returns the expires key
+     */
+    public String expiresKey() {
+        if (this.isCustom) {
+            throw SdkClientException.create("This is a cookie with a custom policy, use policy instead of expires.");
+        }
+        return EXPIRES_KEY;
+    }
+
+    /**
+     * Returns the policy key
+     */
+    public String policyKey() {
+        if (!this.isCustom) {
+            throw SdkClientException.create("This is a cookie with a canned policy, use expires instead of policy");
+        }
+        return POLICY_KEY;
+    }
+
+    /**
+     * Returns the key pair ID value
+     */
+    public String keyPairIdVal() {
+        return keyPairIdVal;
+    }
+
+    /**
+     * Returns the signature value
+     */
+    public String signatureVal() {
+        return signatureVal;
+    }
+
+    /**
+     * Returns the expires value
+     */
+    public String expiresVal() {
+        if (this.isCustom) {
+            throw SdkClientException.create("This is a cookie with a custom policy, use policy instead of expires.");
+        }
+        return expiresVal;
+    }
+
+    /**
+     * Returns the policy value
+     */
+    public String policyVal() {
+        if (!this.isCustom) {
+            throw SdkClientException.create("This is a cookie with a canned policy, use expires instead of policy");
+        }
+        return policyVal;
+    }
+
+    /**
+     * Returns true if this is a cookie with custom policy, else false for a cookie with canned policy
+     */
+    public boolean isCustom() {
+        return isCustom;
+    }
+
+    /**
+     * Returns the cookie header value for the specified cookie type.
+     *
+     * @param type The type of the cookie header value to retrieve
+     * @return The cookie header value to pass into an HTTP request
+     */
+    public String cookieHeaderValue(String type) {
+        switch (type) {
+            case "KeyPairId":
+                return KEY_PAIR_ID_KEY + "=" + keyPairIdVal;
+            case "Signature":
+                return SIGNATURE_KEY + "=" + signatureVal;
+            case "Expires":
+                if (this.isCustom) {
+                    throw SdkClientException.create("This is a cookie with a custom policy, use policy instead of expires.");
+                }
+                return EXPIRES_KEY + "=" + expiresVal;
+            case "Policy":
+                if (!this.isCustom) {
+                    throw SdkClientException.create("This is a cookie with a canned policy, use expires instead of policy");
+                }
+                return POLICY_KEY + "=" + policyVal;
+            default:
+                throw SdkClientException.create("Did not provide a valid cookie type");
+        }
+    }
+
+    @NotThreadSafe
+    @SdkPublicApi
+    public interface Builder extends CopyableBuilder<CloudFrontSignedCookie.Builder, CloudFrontSignedCookie> {
```

Comment:
Combining the two types of cookies in one class comes with some awkward usage, since you never really know whether to use expires or policy. 

Consider creating a set of interfaces to represent a signed cookie, a canned signed cookie and a custom signed cookie and then add two internal default classes to represent the two implementations. In this case the main interface could define all public getters plus a builder interface. 

The benefit is that you don't need to check the isCustoms flag, and you could even have a toSdkHttpRequest method that can generate the whole request for you

Reply:
Implemented interfaces and default classes